### DISCUSSION TITLE
Is there a way for MDX to evaluate components inside ``` blocks?

### DISCUSSION THREAD

**Discussion by behrangsa** [mdx-js/mdx#2165] [Q&A] [answered]
Hi,

If I have a component like this:

```tsx
const Error = ({ children }) => {
  return <span className={styles.error}>{children}</span>;
};
```

Is there a way I can use it inside a _```_ block?

```mdx

## Here is an example:

    ```
    <Error>val s =</Error>
    ```

```

**Comment by wooorm** [comment, marked as answer]
No. MDX is intended to not evaluate code blocks. Similar to how a code block in markdown doesn’t get evaluated.
We use expressions:

```markdown
{
  1 + 2
}
```

However, everything is possible with plugins: https://github.com/mdx-js/mdx/issues/701#issuecomment-947030041.
